server/content.js

```diff
@@ -2,14 +2,56 @@ const fse = require('fs-extra');
 const fs = require('fs');
 const path = require('path');
 
+const fileExistError = new Error('This slug name is already taken');
+
 const save = (contentType, contentName, data, cb) => {
-  console.log(data);
+  const parsedData = JSON.parse(String(data));
+  const filePath = path.resolve('_content', contentType, contentName + '.json');
+  const newFilePath = path.resolve('_content', contentType, parsedData.slug + '.json');
+
+  if (contentName === parsedData.slug) {
+    fse.outputFile(newFilePath, data, cb);
+  } else {
+    fs.stat(newFilePath, function(err, stat) {
+      if (err === null) {
```

Comment:
`if (!err)` would be better in such situations, this agreement to throw err as a first arg is valid, but in general not all libraries gonna send `null` when no errors